latex/programming_interface.tex

```diff
@@ -3559,13 +3559,13 @@ \section{Synchronization and atomics}
 
-An instance of the SYCL \codeinline{stream} class has a maximum buffer size (\codeinline{bufferSize}) that specifies maximum size of the character stream that can be output by any single work item in characters (a character is of size \codeinline{sizeof(char)} bytes) during a kernel invocation, and a maximum statement size (\codeinline{maxStatementSize}) that specifies the maximum size of the character stream that can be output in a single statement in characters (a character is of size \codeinline{sizeof(char)} bytes).  \codeinline{bufferSize} must be sufficient to contain the characters output by statements during execution of a work item, and \codeinline{maxStatementSize} must be sufficient to contain the characters output by any specific statement, otherwise behavior is undefined.  Unused characters within \codeinline{maxStatementSize} (any portion of the \codeinline{maxStatementSize} that is not used/output by a statement) do not count toward the \codeinline{bufferSize} limit of a work item.
+An instance of the SYCL \codeinline{stream} class has a maximum buffer size (\codeinline{totalBufferSize}) that specifies maximum size of the overall character stream that can be output in characters (a character is of size \codeinline{sizeof(char)} bytes) during a kernel invocation, and a maximum stream size (\codeinline{workItemBufferSize}) that specifies the maximum size of the character stream (number of characters) that can be output within a work item before a flush must be performed (a character is of size \codeinline{sizeof(char)} bytes).  \codeinline{totalBufferSize} must be sufficient to contain the characters output by all stream statements during execution of a kernel invocation (the aggregate of outputs from all work items), and \codeinline{workItemBufferSize} must be sufficient to contain the characters output within a work item between stream flush operations.
 
-All member functions of the \codeinline{stream} class are synchronous and errors are handled by throwing synchronous SYCL exceptions.
+If the \codeinline{totalBufferSize} or \codeinline{workItemBufferSize} limits are exceeded, it is implementation defined whether the streamed characters exceeding the limit are output, or silently ignored/discarded, and if output it is implementation defined whether those extra characters exceeding the \codeinline{workItemBufferSize} limit count toward the \codeinline{totalBufferSize} limit.  Regardless of this implementations defined behavior of output exceeding the limits, no undefined or erroneous behavior is permitted of an implementation when the limits are exceeded.  Unused characters within \codeinline{workItemBufferSize} (any portion of the \codeinline{workItemBufferSize} capacity that has not been used at the time of a stream flush) do not count toward the \codeinline{totalBufferSize} limit, in that only characters flushed count toward the \codeinline{totalBufferSize} limit.
```

Comment:
Yes, correct.  We don't want to mandate that characters beyond the limit are output, but we also don't want to allow crashes or data corruption, because that won't be usable and this is mostly a debug feature.  Do we need to clarify this wording in some way?

Reply:
Great, no I think this is clear, I just wanted to make sure I was interpreting that correctly, thanks.